src/api/ingestion.ts

```diff
@@ -33,6 +33,46 @@ export async function removeReferences(fileNames: string[]) {
   }
 }
 
+const UPDATABLE_FIELDS: (keyof ReferenceItem)[] = ['citationKey', 'title', 'publishedDate', 'authors'];
+function applyPatch(
+  field: keyof ReferenceItem,
+  patch: Partial<ReferenceItem>,
+  fn: (patch: Partial<ReferenceItem>) => Partial<Reference>,
```

Comment:
You can drop the argument here: `fn: () => Partial<Reference>`, you always pass functions that take no arguments when you use `applyPatch`

Reply:
good catch. In the final version I removed the arg.